components/dashboard/src/components/UsageView.tsx

```diff
@@ -36,22 +40,37 @@ function UsageView({ attributionId }: UsageViewProps) {
     const today = new Date();
     const startOfCurrentMonth = new Date(today.getFullYear(), today.getMonth(), 1);
     const timestampStartOfCurrentMonth = startOfCurrentMonth.getTime();
-    const [startDateOfBillMonth, setStartDateOfBillMonth] = useState(timestampStartOfCurrentMonth);
-    const [endDateOfBillMonth, setEndDateOfBillMonth] = useState(Date.now());
+    const [startDate, setStartDate] = useState(timestampStartOfCurrentMonth);
+    const [startDateError, setStartDateError] = useState("");
+    const [endDate, setEndDate] = useState(Date.now());
     const [totalCreditsUsed, setTotalCreditsUsed] = useState<number>(0);
     const [isLoading, setIsLoading] = useState<boolean>(true);
     const [supportedClasses, setSupportedClasses] = useState<SupportedWorkspaceClass[]>([]);
 
+    const location = useLocation();
     useEffect(() => {
+        const match = /#(\d{4}-\d{2}-\d{2}):(\d{4}-\d{2}-\d{2})/.exec(location.hash);
```

Comment:
What's the reason to use a hash and not standard query params such as `from` and `to`?

The Regex looks to only match days, without timestamps. Do we want to allow for time also?

We already use `dayjs` in the project, perhaps using https://day.js.org/docs/en/parse/string-format would be cleaner here?

Reply:
I intentionally opted for has instead of URL params, because this is entirely clientside.

Reply:
> The Regex looks to only match days, without timestamps. Do we want to allow for time also?

Yes, it does only match days. Why would you use time? We don't display the times in the ranges.

Reply:
It would replace the regex also, with the date parse string. Basically, we're writing the regex ourselves when it's already been done for us (and tested). Up to you.

Reply:
> Yes, it does only match days. Why would you use time? We don't display the times in the ranges.

I would encode time as well, for completeness. Yes, it would require a time selection UI, or you could default to midninght on both, unless the URL params say otherwise.

Reply:
> It would replace the regex also, with the date parse string. Basically, we're writing the regex ourselves when it's already been done for us (and tested). Up to you.

It would not replace the regex, because the regex includes the delimiter between `<from>:<to>`.

Reply:
I find it more readable to be strict, but I'm happy to just match for the delimiter and pass what every is on both sides into `Date.parse` and then validate the resulting dates.

Reply:
Maybe even render parsing and validation errors. Although ultimately the protocol is intended to be used by ourselves.

Reply:
What do you mean `strict`? 

It would be used as `dayjs(queryParam, 'YYYY-MM-DD')` - this is exactly as strict as before, but actually communicates which format it's using by using standard date parse arguments.